installer/app/src/views/azure-create-app-tutorial.js.jsx

```diff
@@ -5,163 +5,240 @@ import Sheet from './css/sheet';
+                <button onClick={this.__handleAdvanceTutorialClick}>Walk me through it</button><br />
+                <button onClick={this.__skipTutorial}>Skip tutorial</button>
+            </div>
+        );
+
+        intro = (s.tutorialSlide || s.skipTutorial) ? null : intro;
+        tutorialSlide = (s.tutorialSlide) ? tutorialSlides[s.tutorialSlide] : null;
+
+        if (s.tutorialSlide || s.skipTutorial) {
+            pagination.prev = (s.skipTutorial && !s.tutorialSlide) ? <button id="azureTutPrev" onClick={this.__skipTutorial}>Back</button> : '';
+            pagination.prev = (s.tutorialSlide > 1) ? <button id="azureTutPrev" onClick={this.__handleAdvanceTutorialClick}>Back</button> : pagination.prev;
+            pagination.next = (s.tutorialSlide < 9) ? <button id="azureTutNext" onClick={this.__handleAdvanceTutorialClick}>Next</button> : '';
+            pagination.submit = (s.skipTutorial || s.tutorialSlide === 9) ? <button type="submit" style={{margin: "0 0 0 503px"}}>Authenticate</button> : '';
+        }
+
+        return (
+            <div id={this.state.styleEl.id}>
+                {intro}
+                {tutorialSlide}
+
+                <div id="azure-tutorial-inputs">
+                    <label for="redirectURI" style={inputStyles.redirectURI}>Redirect URI</label>
+                    <input name="redirectURI" type="text" value={redirectURI} onClick={this.__handleRedirectURIInputClick}  style={inputStyles.redirectURI} />
+                    <label for="client_id" style={inputStyles.clientId}>App Client ID</label>
+                    <input name="client_id" type="text" placeholder="ab7c1052-1fe7-4642-91f6-065c94de25d4" style={inputStyles.clientId} />
+                    <label for="endpoint" style={inputStyles.endpoint}>OAuth 2.0 Token Endpoint</label>
+                    <input name="endpoint" type="text" placeholder="https://login.microsoftonline.com/{your-uid}/oauth2/token?api-version=1.0" style={inputStyles.endpoint} />
+                </div>
+
+                {pagination.prev} {pagination.next} {pagination.submit}
+            </div>
+        );
+    },
+
+    componentDidMount: function () {
+        this.state.styleEl.commit();
+    },
+
+    /**
+     * Modifies the state's `tutorialSlide` variable, either advancing or stepping back in the
+     * tutorial's walkthrough.
+     * @param  {event} e
+     */
+    __handleAdvanceTutorialClick: function (e) {
+        var s = this.state;
+
+        // Validate Inputs
+        if (s.tutorialSlide === 5 && document.querySelector('[name=client_id]').value === '') {
+            document.querySelector('[name=client_id]').focus();
+            return;
+        } else if (s.tutorialSlide === 8 && document.querySelector('[name=endpoint]').value === '') {
+            document.querySelector('[name=endpoint]').focus();
+            return;
+        }
+
+        if (!s.tutorialSlide || s.tutorialSlide >= 9) {
```

Comment:
Use the `length` property of `tutorialSlides` once it's defined outside the scope of the class instead of `9`.